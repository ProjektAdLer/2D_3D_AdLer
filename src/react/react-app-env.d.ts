/// <reference types="react-scripts" />

// Images
declare module '*.jpg';
declare module '*.png';
declare module '*.env';

// 3D types
declare module '*.glb';
declare module '*.stl';

declare module '*.jpeg';
declare module '*.gif';
declare module '*.mp4';
