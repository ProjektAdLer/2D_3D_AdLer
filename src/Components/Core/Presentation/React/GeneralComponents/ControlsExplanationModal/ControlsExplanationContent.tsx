import { useTranslation } from "react-i18next";
import tailwindMerge from "../../../Utils/TailwindMerge";
import { AdLerUIComponent } from "src/Components/Core/Types/ReactTypes";

import leftClickIcon from "src/Assets/icons/left-click.svg";
import middleMouseIcon from "src/Assets/icons/middle-mouse-click.svg";
import middleMouseDragIcon from "src/Assets/icons/middle-mouse-click-and-drag.svg";
import touchIcon from "src/Assets/icons/touch.svg";
import touchDragIcon from "src/Assets/icons/touch-and-drag.svg";
import touchZoomIcon from "src/Assets/icons/touch-zoom.svg";
import hintIcon from "src/Assets/icons/hint.svg";

import LearningElementUnsolvedIcon from "src/Assets/icons/HighlightColors/LearningElementUnsolvedIcon.svg";
import LearningElementSolvedIcon from "src/Assets/icons/HighlightColors/LearningElementSolvedIcon.svg";
import NonLearningElementBaseIcon from "src/Assets/icons/HighlightColors/NonLearningElementBaseIcon.svg";

export default function ControlsExplanationContent({
  className,
}: AdLerUIComponent<{}>) {
  const { t: translate } = useTranslation("controls");
  return (
    <div
      className={tailwindMerge(
        className,
        "grid grid-rows-6 grid-flow-row gap-x-8 gap-y-3 pb-4 text-xs max-w-[85vw] mobile-landscape:justify-center mobile-landscape:grid-cols-1 mobile-landscape:grid-rows-auto mobile-landscape:grid-flow-row mobile-landscape:gap-y-3 mobile-landscape:pb-0 mobile-landscape:text-2xs mobile-landscape:gap-x-1 portrait:grid-flow-row portrait:grid-cols-1 portrait:grid-rows-auto portrait-gap-y-1 lg:text-sm xl:gap-2",
      )}
    >
      {/* Avatar Controls */}
      <h3 className="self-center font-bold text-adlerdarkblue">
        {translate("controlAvatar")}
      </h3>
      <div className="flex flex-row items-start justify-start gap-2">
        <img className="w-6" alt="Linksklick" src={leftClickIcon}></img>
        <p>{translate("controlAvatarMouse")}</p>
      </div>
      <div className="flex flex-row items-start justify-start gap-2">
        <img className="w-6" alt="Einfacher Touch" src={touchIcon}></img>
        <p>{translate("controlAvatarTouch")}</p>
      </div>
      <div className="flex flex-row items-start justify-start gap-2 portrait:pb-3 ">
        <img className="w-6" alt="Icon" src={hintIcon}></img>
        <p>{translate("controlAvatarHint")}</p>
      </div>

      {/* Open Content Controls */}
      <h3 className="self-center font-bold text-adlerdarkblue">
        {translate("openContent")}
      </h3>
      <div className="flex flex-row items-start justify-start gap-2">
        <img className="w-6" alt="Icon" src={leftClickIcon}></img>
        <p>{translate("openContentMouse")}</p>
      </div>
      <div className="flex flex-row items-start justify-start gap-2">
        <img className="w-6" alt="Icon" src={touchIcon}></img>
        <p>{translate("openContentTouch")}</p>
      </div>
      <div className="flex flex-row items-start justify-start gap-2 portrait:pb-3">
        <img className="w-6" alt="Icon" src={hintIcon}></img>
        <div className="flex flex-col gap-1">
          <p>{translate("openContentHint1")}</p>
          <p>{translate("openContentHint2")}</p>
        </div>
      </div>

      {/* Rotate Camera Controls */}
      <h3 className="self-center font-bold text-adlerdarkblue">
        {translate("rotateCamera")}
      </h3>
      <div className="flex flex-row items-start justify-start gap-2">
        <img className="w-6" alt="Icon" src={middleMouseDragIcon}></img>
        <p>{translate("rotateCameraMouse")}</p>
      </div>
      <div className="flex flex-row items-start justify-start gap-2">
        <img className="w-6" alt="Icon" src={touchDragIcon}></img>
        <p>{translate("rotateCameraTouch")}</p>
      </div>
      <div className="flex flex-row items-start justify-start gap-2 portrait:pb-3">
        <img className="w-6" alt="Icon" src={hintIcon}></img>
        <p>{translate("rotateCameraHint")}</p>
      </div>

      {/* Zoom Camera Controls */}
      <h3 className="self-center font-bold text-adlerdarkblue">
        {translate("zoomCamera")}
      </h3>
      <div className="flex flex-row items-start justify-start gap-2">
        <img className="w-6" alt="Icon" src={middleMouseIcon}></img>
        <p>{translate("zoomCameraMouse")}</p>
      </div>
      <div className="flex flex-row items-start justify-start gap-2">
        <img className="w-6" alt="Icon" src={touchZoomIcon}></img>
        <p>{translate("zoomCameraTouch")}</p>
      </div>
      <div className="flex flex-row items-start justify-start gap-2 portrait:pb-3">
        <img className="w-6" alt="Icon" src={hintIcon}></img>
        <p>{translate("zoomCameraHint")}</p>
      </div>

      {/* Highlight Colors */}
      <h3 className="self-center font-bold text-adlerdarkblue">
        {"Highlight Farben"}
      </h3>
      <div className="flex flex-row items-start justify-start gap-2">
        <img className="w-6" alt="Icon" src={LearningElementUnsolvedIcon}></img>
        <p>{translate("LearningElementUnsolvedIcon")}</p>
      </div>
      <div className="flex flex-row items-start justify-start gap-2">
        <img className="w-6" alt="Icon" src={LearningElementSolvedIcon}></img>
        <p>{translate("LearningElementSolvedIcon")}</p>
      </div>
      <div className="flex flex-row items-start justify-start gap-2">
        <img className="w-6" alt="Icon" src={NonLearningElementBaseIcon}></img>
        <p>{translate("NonLearningElementBaseIcon")}</p>
      </div>
      <div className="flex flex-row items-start justify-start gap-2">
        <img className="w-6" alt="Icon" src={hintIcon}></img>
        <p>{translate("HighlightHint")}</p>
      </div>
    </div>
  );
}
