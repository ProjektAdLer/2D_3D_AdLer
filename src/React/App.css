body {
  margin: 0;
  height: 100vh;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

div{
  width: 100%;
  height: 100%;
}

section{
  display: inline;
  position: fixed;
  top: 0;
  left: 0;
  width: 120px;
  height: 100vh;
  background-color: #fff;
}

button{
  width: 100px;
  margin: 10px;
  z-index: 5;
  position: fixed;
  left: 0;
  top: 0;
}

canvas {
  width: calc(100% - 100px);
  height: 100%;
  margin-left: 100px;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}

.modal {
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background: rgba(0, 0, 0, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
}

.modal-content {
  width: 1000px;
  background-color: #fff;
}

.modal-header,
.modal-footer {
  padding: 10px;
  height: 60px;
}

.modal-title {
  margin: 0;
}

.modal-body {
  padding: 10px;
  border-top: 1px solid #eee;
  border-bottom: 1px solid #eee;
}
