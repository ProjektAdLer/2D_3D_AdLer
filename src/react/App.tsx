import React from 'react';
import './App.css';
import BabylonCanvas from './components/BabylonCanvas';

function App() {
	return (
		<React.Fragment>
			<BabylonCanvas style={{ width: '100%', position: 'absolute' }} />
		</React.Fragment>
	);
}

export default App;
