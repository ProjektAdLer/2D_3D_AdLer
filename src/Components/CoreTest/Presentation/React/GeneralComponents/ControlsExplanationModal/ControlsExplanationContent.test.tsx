import React from "react";
import { render } from "@testing-library/react";
import ControlsExplanationContent from "../../../../../Core/Presentation/React/GeneralComponents/ControlsExplanationModal/ControlsExplanationContent";

describe("ControlsExplanationContent", () => {
  test("should render", () => {
    const { container } = render(<ControlsExplanationContent />);
    expect(container).toMatchSnapshot();
  });
});
