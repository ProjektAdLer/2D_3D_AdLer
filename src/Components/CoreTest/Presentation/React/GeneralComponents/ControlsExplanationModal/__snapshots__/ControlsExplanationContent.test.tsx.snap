// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlsExplanationContent should render 1`] = `
<div>
  <div
    class="grid landscape:grid-rows-6 landscape:grid-flow-col portrait:grid-flow-row portrait:grid-cols-1 portrait:grid-rows-auto gap-x-8 landscape:gap-y-3 portrait:gap-6 pb-4 text-xs lg:text-sm"
  >
    <h3
      class="self-center font-bold text-adlerdarkblue"
    >
      controlAvatar
    </h3>
    <div
      class="flex flex-row items-start justify-start gap-2 min-h-10 portrait:min-h-0"
    >
      <img
        alt="Linksklick"
        class="w-6"
        src="left-click.svg"
      />
      <p>
        controlAvatarMouse
      </p>
    </div>
    <div
      class="flex flex-row items-start justify-start gap-2"
    >
      <img
        alt="Einfacher Touch"
        class="w-6"
        src="touch.svg"
      />
      <p>
        controlAvatarTouch
      </p>
    </div>
    <div
      class="flex flex-row items-start justify-start gap-2 landscape:row-span-3 portrait:pb-6"
    >
      <img
        alt="Icon"
        class="w-6"
        src="hint.svg"
      />
      <p>
        controlAvatarHint
      </p>
    </div>
    <h3
      class="self-center font-bold text-adlerdarkblue"
    >
      openContent
    </h3>
    <div
      class="flex flex-row items-start justify-start gap-2 min-h-10 portrait:min-h-0"
    >
      <img
        alt="Icon"
        class="w-6"
        src="left-click.svg"
      />
      <p>
        openContentMouse
      </p>
    </div>
    <div
      class="flex flex-row items-start justify-start gap-2"
    >
      <img
        alt="Icon"
        class="w-6"
        src="touch.svg"
      />
      <p>
        openContentTouch
      </p>
    </div>
    <div
      class="flex flex-row items-start justify-start gap-2 landscape:row-span-3 portrait:pb-6"
    >
      <img
        alt="Icon"
        class="w-6"
        src="hint.svg"
      />
      <div
        class="flex flex-col gap-1"
      >
        <p>
          openContentHint1
        </p>
        <p>
          openContentHint2
        </p>
      </div>
    </div>
    <h3
      class="self-center font-bold text-adlerdarkblue"
    >
      rotateCamera
    </h3>
    <div
      class="flex flex-row items-start justify-start gap-2 min-h-10 portrait:min-h-0"
    >
      <img
        alt="Icon"
        class="w-6"
        src="middle-mouse-click-and-drag.svg"
      />
      <p>
        rotateCameraMouse
      </p>
    </div>
    <div
      class="flex flex-row items-start justify-start gap-2"
    >
      <img
        alt="Icon"
        class="w-6"
        src="touch-and-drag.svg"
      />
      <p>
        rotateCameraTouch
      </p>
    </div>
    <div
      class="flex flex-row items-start justify-start gap-2 landscape:row-span-3 portrait:pb-6"
    >
      <img
        alt="Icon"
        class="w-6"
        src="hint.svg"
      />
      <p>
        rotateCameraHint
      </p>
    </div>
    <h3
      class="self-center font-bold text-adlerdarkblue"
    >
      zoomCamera
    </h3>
    <div
      class="flex flex-row items-start justify-start gap-2 min-h-10 portrait:min-h-0"
    >
      <img
        alt="Icon"
        class="w-6"
        src="middle-mouse-click.svg"
      />
      <p>
        zoomCameraMouse
      </p>
    </div>
    <div
      class="flex flex-row items-start justify-start gap-2"
    >
      <img
        alt="Icon"
        class="w-6"
        src="touch-zoom.svg"
      />
      <p>
        zoomCameraTouch
      </p>
    </div>
    <div
      class="flex flex-row items-start justify-start gap-2 landscape:row-span-3 portrait:pb-6"
    >
      <img
        alt="Icon"
        class="w-6"
        src="hint.svg"
      />
      <p>
        zoomCameraHint
      </p>
    </div>
    <h3
      class="self-center font-bold text-adlerdarkblue"
    >
      Highlight Farben
    </h3>
    <div
      class="flex flex-row items-start justify-start gap-2 min-h-10 portrait:min-h-0"
    >
      <img
        alt="Icon"
        class="w-6"
        src="LearningElementUnsolvedIcon.svg"
      />
      <p>
        LearningElementUnsolvedIcon
      </p>
    </div>
    <div
      class="flex flex-row items-start justify-start gap-2"
    >
      <img
        alt="Icon"
        class="w-6"
        src="LearningElementSolvedIcon.svg"
      />
      <p>
        LearningElementSolvedIcon
      </p>
    </div>
    <div
      class="flex flex-row items-start justify-start gap-2"
    >
      <img
        alt="Icon"
        class="w-6"
        src="NonLearningElementBaseIcon.svg"
      />
      <p>
        NonLearningElementBaseIcon
      </p>
    </div>
    <div
      class="flex flex-row items-start justify-start gap-2"
    >
      <img
        alt="Icon"
        class="w-6"
        src="hint.svg"
      />
      <p>
        HighlightHint
      </p>
    </div>
  </div>
</div>
`;
